tests/integration/CapabilityTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Automattic\LegacyRedirector\Tests;
+
+use Automattic\LegacyRedirector\Capability;
+
+/**
+ * CapabilityTest class
+ */
+class CapabilityTest extends TestCase {
+
+	/**
+	 * Test capabilities for Capability::MANAGE_REDIRECTS_CAPABILITY
+	 *
+	 * @return void
+	 */
+	public function test_new_admin_capability() {
+
+		$capability = new Capability();
+
+		// We need to force clear capabilities here as the wp_options `roles` option might not get cleared after a failed test.
+		$capability->unregister();
+
+		// in WP_User class, if multisite and user is administrator, all capabilities are allowed, so this test is not useful.
+		if ( ! is_multisite() ) {
+			// We check if a new admin user does not have the redirect capability.
+			$user_id = 1;
+			$this->assertUserNotHasRedirectCapability( $user_id );
+		}
+
+		$this->assertRoleNotHasRedirectsCapability( 'administrator' );
+		$this->assertRoleNotHasRedirectsCapability( 'editor' );
+		$this->assertRoleNotHasRedirectsCapability( 'subscriber' );
+		$this->assertRoleNotHasRedirectsCapability( '' );
+
+		$capability->register();
+
+		$this->assertRoleHasRedirectsCapability( 'administrator' );
+		$this->assertRoleHasRedirectsCapability( 'editor' );
+		$this->assertRoleNotHasRedirectsCapability( 'subscriber' ); // Should be no change.
+		$this->assertRoleNotHasRedirectsCapability( '' ); // Should be no change.
+	}
+
+	/**
+	 * Test the Capability unregister method
+	 *
+	 * @return void
+	 */
+	public function test_register() {
+		$capability = new Capability();
+		$capability->register();
+
+		$this->assertFalse( $capability->register() );
+
+		$this->assertRoleHasRedirectsCapability( 'administrator' );
+
+		$capability->unregister();
+
+		$this->assertRoleNotHasRedirectsCapability( 'administrator' );
+
+		$this->assertTrue( $capability->register() );
+	}
+
+	/**
+	 * Test if the capabilities register function fires again after init
+	 *
+	 * @return void
+	 */
+	public function tearDown() {
+
+		$capability = new Capability();
+		$capability->unregister();
+	}
```

Comment:
Move this fixture to the top, and check the DocBlock description.

Reply:
fixed